opae-libs/plugins/xfpga/metrics/metrics_utils.c

```diff
@@ -692,7 +692,6 @@ fpga_result get_bmc_metrics_values(fpga_handle handle,
 		_handle->_bmc_metric_cache_value = calloc(sizeof(struct _fpga_bmc_metric), num_sensors);
 		if (_handle->_bmc_metric_cache_value == NULL) {
 			OPAE_ERR("Failed to allocate memory");
-			result = FPGA_NO_MEMORY;
 			goto out_destroy;
```

Comment:
Doesn't removing
```c
result = FPGA_NO_MEMORY;
```
cause a stale result to be returned from the function?

Reply:
The out_destroy: block

out_destroy:                                                                                                                  
        result = xfpga_bmcDestroySDRs(_handle, &records);                                                                     
        if (result != FPGA_OK) {                                                                                              
                OPAE_ERR("Failed to Destroy SDR.");                                                                           
                return result;                                                                                                
        } 
return result;

the first result is overwritten by the later second.